### ISSUE TITLE
LabelList Position Incorrect for 2.0.0-beta.5

### ISSUE THREAD

**Issue by epihel** [recharts/recharts#2119]
- [x] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/recharts-issue-template-ft38g)

### Steps to reproduce
1) Create a horizontal BarChart with a Bar and a LabelList with the label position "right:"

```
<BarChart data={data} height={300} layout="vertical" width={600}>
    <Bar barSize={ 20 } dataKey="value" fill="rgb(72, 142, 151)">
        <LabelList dataKey="value" position="right" />
    </Bar>
</BarChart>
```

### What is expected?
The labels should be positioned to the right of each bar.

### What is actually happening?
The labels appear on the inside left of each bar.

| Environment | Info |
|---|---|
| Recharts | v2.0.0-beta.5 |
| React | 16.13.1 |
| System | OSX 10.13.4 (High Sierra) |
| Browser | Chrome 81.0.4044.129 |

---
The same code in version 1.8.5 produces the expected results.

**Comment by dillonreedy** [comment]
Hi there,

If you update your codesandbox to [beta version 8](https://github.com/recharts/recharts/releases/tag/v2.0.0-beta.8) this issue is no longer occurring.

At present the current v2 release won't work though.

**Comment by SirZach** [comment]
Hi @dillonreedy not sure if expected, but I'm on version `2.1.6` and still experiencing this issue. Thanks!

**Comment by semoal** [comment]
Without a reproduction repo we can't work on it guys. 

Thanks

**Comment by ckifer** [comment]
this looks like it was fixed in beta.8

it is fixed (or at least the original issue is fixed) in 2.1.6+ as well. 

Closing this as completed - please create a new issue if you're still experiencing this
